websockify/websockifyserver.py

```diff
@@ -700,6 +705,17 @@ def start_server(self):
 
         if self.listen_fd != None:
             lsock = socket.fromfd(self.listen_fd, socket.AF_INET, socket.SOCK_STREAM)
+        elif self.listen_sock != None:
+            # Make sure the socket does not already exist
+            try:
+                os.unlink(self.listen_sock)
+            except OSError:
+                if os.path.exists(self.listen_sock):
+                    raise
+            lsock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+            lsock.bind(self.listen_sock)
+            os.chmod(self.listen_sock, self.listen_sock_mode)
+            lsock.listen(100)
         else:
             lsock = self.socket(self.listen_host, self.listen_port, False,
```

Comment:
I think you should probably move the socket setup in to this function. It already handles Unix sockets, just not listening ones.

Reply:
Can do, I´ve seen the method was easier to put it in there at first.

Not entierly sure about the ```lsock.listen(100)```. I took the 100 from somewhere else in the code. But unsure why it would be needed. Probably not necessary to pass anything to the constructor here.

Reply:
You got that 100 from the socket setup function. So things would definitely clearer with all `listen()` calls in the same place. :)